File: python/1061.py
_, startDay = input().split()
startTime = input().split(" : ")
_, endDay = input().split()
endTime = input().split(" : ")

startDay = int(startDay)

endDay = int(endDay)

startHour = int(startTime[0])
startMinutes = int(startTime[1])
startSeconds = int(startTime[2])

endHour = int(endTime[0])
endMinutes = int(endTime[1])
endSeconds = int(endTime[2])

totalDays = 0
totalHours = 0
totalMinutes = 0
totalSeconds = 0

totalDays = endDay - startDay

if endHour > startHour or (endHour == startHour and endMinutes > startMinutes) or (endHour == startHour and endMinutes == startMinutes and endSeconds > startSeconds):
    totalHours = endHour - startHour
    totalMinutes = endMinutes - startMinutes
    totalSeconds = endSeconds - startSeconds

if endHour < startHour:
    totalDays -= 1
    totalHours = 24 - (startHour - endHour)
else: 
    totalHours = endHour - startHour
    
if endMinutes < startMinutes:
    if totalHours <= 0:
        totalDays -= 1
        totalHours = 24 - (totalHours) -1
    else:
        totalHours -= 1
    totalMinutes =  60 - (startMinutes - endMinutes)
else:
    totalMinutes =  endMinutes - startMinutes
    
if endSeconds < startSeconds:
    if totalMinutes <= 0:
        totalHours -= 1
        totalMinutes = 60 - (totalMinutes) - 1
    else:
        totalMinutes -= 1
    totalSeconds = 60 - (startSeconds - endSeconds)
else:
    totalSeconds = endSeconds - startSeconds
    
print(f'{totalDays} dia(s)')
print(f'{totalHours} hora(s)')
print(f'{totalMinutes} minuto(s)')
print(f'{totalSeconds} segundo(s)')

File: lua/1061.lua
local startDay = io.read()
local startTime = io.read()

local tempTable = {}
startDay = string.match(startDay, '%d+')
for n in string.gmatch(startTime, '%d+') do
    table.insert(tempTable, n)
end

local startHour = tonumber(tempTable[1])
local startMinute = tonumber(tempTable[2])
local startSecond = tonumber(tempTable[3])

tempTable = {}

local endDay = io.read()
local endTime = io.read()

endDay = string.match(endDay, '%d+')
for n in string.gmatch(endTime, '%d+') do
    table.insert(tempTable, n)
end

local endHour = tonumber(tempTable[1])
local endMinute = tonumber(tempTable[2])
local endSecond = tonumber(tempTable[3])

local days = endDay - startDay
local hours = endHour - startHour
local minutes = endMinute - startMinute
local seconds = endSecond - startSecond

if seconds < 0 then
    seconds = 60 + seconds
    minutes = minutes - 1
end
if minutes < 0 then
    minutes = 60 + minutes
    hours = hours - 1
end
if hours < 0 then
    hours = 24 + hours
    days = days - 1
end

print(days.." dia(s)")
print(hours.." hora(s)")
print(minutes.." minuto(s)")
print(seconds.." segundo(s)")